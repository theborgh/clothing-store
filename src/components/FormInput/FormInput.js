import React from 'react';
import './FormInput.scss'

const formInput = ({ handleChange, label, ...otherProps }) => (

   <div className="group" data-test="component-FormInput" >
      <input className="formInput" onChange={handleChange} {...otherProps} />
      {
         label ? 
         <label className={`${otherProps.value.length ? 'shrink' : ''} formInputLabel`} >
            {label}
         </label> : null
      }
   </div>
);

export default formInput;