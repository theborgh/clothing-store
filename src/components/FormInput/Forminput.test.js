import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import FormInput from './FormInput';

configure({ adapter: new Adapter() });

const createShallowFormInputWrapper = (props={}, state=null) => {
  const wrapper = shallow(<FormInput {...props} />);

  if (state) {
    wrapper.setState(state);
  }

return wrapper;
}
const findTagsWithTestAttribute = (wrapper, attributeValue) => wrapper.find(`[data-test='component-${attributeValue}']`);

it("renders without errors", () => {
  const wrapper = createShallowFormInputWrapper();
  expect(findTagsWithTestAttribute(wrapper, 'component-FormInput').length).toBe(1);
});

it("matches the snapshot", () => {
  expect(shallow(<FormInput />).debug()).toMatchSnapshot();
})