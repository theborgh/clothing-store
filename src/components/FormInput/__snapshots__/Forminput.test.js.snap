// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
"<div className=\\"group\\" data-test=\\"component-FormInput\\">
  <input className=\\"formInput\\" onChange={[undefined]} />
</div>"
`;
